spa/templates/basic-auth.js

```diff
@@ -1,26 +1,29 @@
 // Expected Authorization header value
 const authorization = "Basic ${credentials}"
 
+// List of regular expressions describing paths excluded from the basic auth
+const exclusions = "${exclusions}".split(" ").filter(function (e) {return e})
+
 exports.handler = (event, context, callback) => {
   // Get request and request headers
   const request = event.Records[0].cf.request
   const headers = request.headers
 
-  // Require Basic authentication
+  // Require Basic authentication unless the exclusions include the request path
   if (
-    !headers.authorization ||
-    headers.authorization[0].value !== authorization
+    exclusions.some(e => (new RegExp(e)).test(request.uri)) ||
```

Comment:
I'd suggest building the regular expressions once, eg.

```js
const exclusions = exclusionPatterns.map(pattern => new RegExp(pattern))
```

and then

```js
exclusions.some(re => re.test(request.uri))
```

Reply:
Optimization and clarification 👍🏻